modules/shells/posix.nix

```diff
@@ -44,11 +44,17 @@ in
       );
 
       exportedEnvVars = let
-        # make foo = "bar" -> foo = ["bar"]
-        allValuesLists =
-          mapAttrs (n: toList) config.environmentVariables;
         exportVariables =
-          mapAttrsToList (n: v: ''export ${n}=${escapeShellArg (concatStringsSep ":" v)}'') allValuesLists;
+          if builtins.isList config.environmentVariables
+          then let
+            # double quote and replace " with \"
+            escapeShellArgToEval = arg: "\"${lib.replaceStrings [''"''] [''\"''] arg}\"";
```

Comment:
@limeytexan your PR replaced ' with '\'', but don't we instead need to replace " with \"? Since we're double quoting instead of single quoting like `escapeShellArg`?